resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -492,10 +492,8 @@ private[spark] class Client(
   private[yarn] def getPreloadedStatCache(
       fsLookup: URI => FileSystem = FileSystem.get(_, hadoopConf)): HashMap[URI, FileStatus] = {
     val statCache = HashMap[URI, FileStatus]()
-    val jars = sparkConf.get(SPARK_JARS)
-    val directoryToFiles = jars.map(directoriesToBePreloaded).getOrElse(HashMap.empty)
-
-    directoryToFiles.foreach { case (dir: URI, filesInDir: HashSet[String]) =>
+    val jars = sparkConf.get(JARS_TO_DISTRIBUTE)
```

Comment:
This does not look right - why are we changing the config used ?

Reply:
JARS_TO_DISTRIBUTE = ConfigBuilder("spark.yarn.dist.jars")
SPARK_JARS = ConfigBuilder("spark.yarn.jars")

So, most user-related jars come from `JARS_TO_DISTRIBUTE`.

Reply:
I am not sure I understood the comment :-)
`statCache` is used within `prepareLocalResources` - within that method, it is primarily used in context of `SPARK_JARS` - so why change to a different config ?

Reply:
Note - it is fine to ignore one-off cases, like app-jar, ivy, etc.

Reply:
Yeah, Actually I was thinking the same thing, maybe should include all those (`SPARK_JARS`, `JARS_TO_DISTRIBUTE`, `FILES_TO_DISTRIBUTE`, `ARCHIVES_TO_DISTRIBUTE`, `pySparkArchives`, `PY_FILES`).